### PR TITLE
Base env cleanup

### PR DESCRIPTION
**Pull request by AboudyKreidieh**
changes:
- removed some traci calls from base_env
- removed some things in base_env that are never used or called

### CODE REVIEW

File: flow/envs/base_env.py [hunk 1]
```diff
@@ -349,8 +314,20 @@ def setup_initial_state(self):
         self.vehicles.update(vehicle_obs, id_lists, self)
         self.traffic_lights.update(tls_obs)
 
-        # store the network observations in the vehicles class
-        self.vehicles.update(vehicle_obs, id_lists, self)
+        # check to make sure all vehicles have been spawned
+        if len(self.initial_ids) < self.vehicles.num_vehicles:
+            logging.error("Not enough vehicles have spawned! Bad start?")
+            exit()
```
Comment (42B) [inline flow/envs/base_env.py]: Shouldn't this use `sys.exit` and return a nonzero value?

Comment (eugenevinitsky) [reply flow/envs/base_env.py]: This is correct, thanks for the catch!

Comment (42B) [reply flow/envs/base_env.py]: @eugenevinitsky 

No problem!

File: flow/envs/base_env.py [hunk 2]
```diff
@@ -323,12 +323,6 @@ def setup_initial_state(self):
             value = sparse state information (only what is needed to add a
             vehicle in a sumo network with traci)
         """
-        # check to make sure all vehicles have been spawned
-        num_spawned_veh = self.traci_connection.simulation.getDepartedNumber()
```
Comment (eugenevinitsky) [inline flow/envs/base_env.py]: why is this a bad check to have in place?

File: flow/envs/multiagent_env.py [hunk 3]
```diff
@@ -261,7 +254,7 @@ def reset(self):
                     route_id,
                     typeID=str(type_id),
                     departLane=str(lane_index),
```
Comment (eugenevinitsky) [inline flow/envs/multiagent_env.py]: Thank you for copying this!

File: flow/envs/base_env.py (hunk 2 above)
Comment (AboudyKreidieh) [reply flow/envs/base_env.py]: Don’t worry, it was just shifted down. The check is still there.

**Pull request merged**